token-metadata/program/tests/update.rs

```diff
@@ -161,4 +171,286 @@ mod update {
 
         assert_eq!(metadata.programmable_config, None);
     }
+
+    #[tokio::test]
+    async fn update_invalid_rule_set() {
+        // Currently users can add an invalid rule set to their pNFT which will effectively
+        // prevent it from being updated again because it either won't be owned by the mpl-token-auth rules
+        // program or it won't be a valid rule set to call validate on.
+        // We relax the check a little to let users fix invalid rule sets.
+        let mut program_test = ProgramTest::new("mpl_token_metadata", mpl_token_metadata::ID, None);
+        program_test.add_program("mpl_token_auth_rules", mpl_token_auth_rules::ID, None);
+        let context = &mut program_test.start_with_context().await;
+
+        let invalid_rule_set = Pubkey::new_unique();
+
+        let authority = Keypair::from_bytes(&context.payer.to_bytes()).unwrap();
+
+        let (authorization_rules, _auth_data) =
+            create_default_metaplex_rule_set(context, authority).await;
+
+        let update_authority = Keypair::from_bytes(&context.payer.to_bytes()).unwrap();
+
+        // Create with an empty rule set so we can test updating.
+        let mut da = DigitalAsset::new();
+        da.create_and_mint(
+            context,
+            TokenStandard::ProgrammableNonFungible,
+            None,
+            None,
+            1,
+        )
+        .await
+        .unwrap();
+
+        let metadata = da.get_metadata(context).await;
+
+        if let Some(ProgrammableConfig::V1 { rule_set }) = metadata.programmable_config {
+            assert_eq!(rule_set, None);
+        }
```

Comment:
Nit: for the test shouldn't this be more like:
```
let config = metadata.programmable_config.unwrap();
assert_eq!(config.rule_set, None)
```
Rather than ignoring non matching cases?

Or at least panic on else like you do below when it's set?

Reply:
Added the `else` statement. It's an enum not a struct so can't read the field directly like you do above.